python/lsst/ts/salobj/base.py

```diff
@@ -126,90 +111,16 @@ def index_impl():
-    def __init__(self, sallib, index=None, debug=0):
-        self.lib = sallib
-        self.name = sallib.componentName[4:]  # lop off leading SAL_
-        if sallib.componentIsMultiple:
-            if index is None:
-                raise RuntimeError(f"Component {self.name} is indexed, so index cannot be None")
-        else:
-            if index not in (0, None):
-                raise RuntimeError(f"Component {self.name} is not indexed so index={index} must be None or 0")
-            index = 0
-        self.index = index
-        Manager = getattr(self.lib, "SAL_" + self.name)
-        self.manager = Manager(self.index)
-        self.manager.setDebugLevel(debug)
-        self._AckType = getattr(self.lib, self.name + "_ackcmdC")
 
-    def __repr__(self):
-        return f"SalInfo({self.name}, {self.index})"
-
-    @property
-    def AckType(self):
-        """The class of command acknowledgement.
-
-        It is contructed with the following parameters
-        and has these fields:
-
-        ack : `int`
-            Acknowledgement code; one of the ``self.lib.SAL__CMD_``
-            constants, such as ``self.lib.SAL__CMD_COMPLETE``.
-        error : `int`
-            Error code; 0 for no error.
-        result : `str`
-            Explanatory message, or "" for no message.
-        """
-        return self._AckType
-
-    def makeAck(self, ack, error=0, result="", truncate_result=False):
-        """Make an AckType object from keyword arguments.
-
-        Parameters
-        ----------
-        ack : `int`
-            Acknowledgement code; one of the ``self.lib.SAL__CMD_``
-            constants, such as ``self.lib.SAL__CMD_COMPLETE``.
-        error : `int`
-            Error code. Should be 0 unless ``ack`` is
-            ``self.lib.SAL__CMD_FAILED``
-        result : `str`
-            More information. This is arbitrary, but limited to
-            `MAX_RESULT_LEN` characters.
-        truncate_result : `bool`
-            What to do if ``result`` is longer than  `MAX_RESULT_LEN`
-            characters:
-
-            * If True then silently truncate ``result`` to `MAX_RESULT_LEN`
-              characters.
-            * If False then raise `ValueError`
-
-        Raises
-        ------
-        ValueError
-            If ``len(result) > `MAX_RESULT_LEN`` and ``truncate_result``
-            is false.
-        """
-        data = self.AckType()
-        data.ack = ack
-        data.error = error
-        if len(result) > MAX_RESULT_LEN:
-            if truncate_result:
-                result = result[0:MAX_RESULT_LEN]
-            else:
-                raise ValueError(f"len(result) > MAX_RESULT_LEN={MAX_RESULT_LEN}; result={result}")
-        data.result = result
-        return data
+def tai_minus_utc(utc):
+    """Return TAI unix seconds, given UTC in unix seconds.
+
+    This function is only intended for current time;
+    it makes no attempt to be correct for historical dates.
+    """
+    return utc + 37
```

Comment:
Should this be also part of DM-19791? Maybe it would be a good idea to add a `#TODO` or a `#FIXME` note.

Reply:
We only need one of the two (and this one is presently broken). I will combine them into one free function named `tai_from_utc` since that is how it is always used.